core/domain/html_cleaner.py

```diff
@@ -245,7 +303,7 @@ def convert_to_text_angular(html_data):
         # In such cases a tag is simply removed.
         elif tag.name == 'a':
             replace_with_link = True
-            if tag.has_attr('href'):
+            if tag.has_attr('href') and tag.get_text():
```

Comment:
Since you added the check for `get_text`, modify the comment above to say `...cases where there is no href attribute, or the link is empty (no text).`

Reply:
Done!